impl/src/main/java/org/ehcache/spi/serialization/DefaultSerializationProvider.java

```diff
@@ -55,9 +55,9 @@
 
   private final TransientProvider transientProvider;
   private final PersistentProvider persistentProvider;
-  
-  final Set<Serializer<?>> created = new HashSet<Serializer<?>>();
-  
+
+  protected final Map<Serializer<?>, AtomicInteger> providedVsCount = new HashMap<Serializer<?>, AtomicInteger>();
```

Comment:
Like for the `CalssInstanceProvider` this needs to become safe for a concurrent environment.